### PR TITLE
UHF-6346: Job listing migration fixes

### PR DESCRIPTION
**Pull request by tapsal**
# [UHF-6346](https://helsinkisolutionoffice.atlassian.net/browse/UHF-6346)
Made changes to the job listing migration.

## What was done
* Take into account translated nodes when unpublishing job listings that are missing from source.
* When importing job listings, skip importing _publish on_ date when the date is in the past and the corresponding node is already published. There is no need to import the date field, as the node state already represents a state after published by scheduler. This change prevents the situation when a past value is imported and the scheduler temporarily updates the node as unpublished.
* Small fix to error: Call to a member function hasField() on null.
* Refactored unpublishing job listings that are missing from source. Now reacting to MigrateMissingSourceRowsEvent so there is no longer e.g. a need to mark all previously imported nodes as ready to be re-imported.

## How to install
* Make sure your instance is up and running on correct branch.
  * `git pull origin UHF-6346-scheduler-publish-changes`
  * `make fresh`
* Run `make drush-cr`

## How to test
* Import job listings for the different languages (using `migrate:import`):
  * `helfi_rekry_jobs:all`
  * `helfi_rekry_jobs:all_sv`
  * `helfi_rekry_jobs:all_en`
* After importing, run cron so that the scheduler publishes and unpublishes nodes.
  * `drush cron`
* Go to the `/admin/content?title=&type=job_listing` page and check the job listings.
* Open the job listing source APIs to your browser.
* Compare the node publish status with the API. Should the nodes be published or not when comparing to the API publicationStarts and publicationEnds values?
  * Also check the different translations.
* Because you run the `make fresh` command, you should have some job listings that are missing from source API. Are those unpublised as they should be?
  * Try publishing some and the run the migrate:import again. Now they should be unpublished again.
* Check that code looks ok.

## Designers review
* [x] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

### CODE REVIEW

**Comment by ltpk** [comment]
Job listings that are published as per their API data (publication start time is in the past and end time in the future) don't have a publication start time in Drupal. Only publication end time is selected. Doesn't seem to affect functionality, though.

File: public/modules/custom/helfi_rekry_content/helfi_rekry_content.module [hunk 1]
```diff
@@ -268,7 +268,7 @@ function helfi_rekry_content_page_attachments_alter(array &$attachments) : void
       // Otherwise, use the one from the taxonomy term.
       $taxonomy_term = $node->field_organization->entity;
 
-      if ($taxonomy_term->hasField('field_default_image') && !$taxonomy_term->field_default_image->isEmpty()) {
+      if (!empty($taxonomy_term) && $taxonomy_term->hasField('field_default_image') && !$taxonomy_term->field_default_image->isEmpty()) {
```
Comment (ltpk) [inline public/modules/custom/helfi_rekry_content/helfi_rekry_content.module:271]: This functionality has been removed from the module (and moved to the theme) in #56

**Review by ltpk — APPROVED** [review]
Works like a charm! 🍺

**Comment by tapsal** [comment]
> Job listings that are published as per their API data (publication start time is in the past and end time in the future) don't have a publication start time in Drupal. Only publication end time is selected. Doesn't seem to affect functionality, though.

This is because Scheduler module clears the field after publishing.

**Pull request merged**